internal/resources/aws/elastic_beanstalk_environment.go

```diff
@@ -0,0 +1,173 @@
+
+	StreamLogs            bool
+	instanceCount         int64
+	MonthlyDataIngestedGB *float64 `infracost_usage:"monthly_data_ingested_gb"`
+	StorageGB             *float64 `infracost_usage:"storage_gb"`
+	MonthlyDataScannedGB  *float64 `infracost_usage:"monthly_data_scanned_gb"`
+	CloudwatchLogGroup    *CloudwatchLogGroup
+	LoadBalancer          *LB
+	ElasticLoadBalancer   *ELB
+	DBInstance            *DBInstance
+}
+
+// ElasticBeanstalkEnvironmentUsageSchema defines a list which represents the usage schema of ElasticBeanstalkEnvironment.
+var ElasticBeanstalkEnvironmentUsageSchema = append([]*schema.UsageItem{
+	{Key: "monthly_data_processed_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "new_connections", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "active_connections", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "rule_evaluations", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "processed_bytes_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "monthly_data_ingested_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "monthly_data_scanned_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "storage_gb", DefaultValue: 0, ValueType: schema.Float64},
+	{Key: "instances", DefaultValue: 0, ValueType: schema.Int64},
+}, InstanceUsageSchema...)
+
+// PopulateUsage parses the u schema.UsageData into the ElasticBeanstalkEnvironment.
+// It uses the `infracost_usage` struct tags to populate data into the ElasticBeanstalkEnvironment.
+func (r *ElasticBeanstalkEnvironment) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+	if r.LoadBalancerType == "classic" {
+		resources.PopulateArgsWithUsage(r.ElasticLoadBalancer, u)
+	} else {
+		resources.PopulateArgsWithUsage(r.LoadBalancer, u)
+	}
+	if r.StreamLogs {
+		resources.PopulateArgsWithUsage(r.CloudwatchLogGroup, u)
+	}
+	if r.RDSIncluded {
+		resources.PopulateArgsWithUsage(r.DBInstance, u)
+	}
+}
+
+// getUsageSchemaWithDefaultInstanceCount is a temporary hack to make --sync-usage-file use the group's "desired_size"
+// as the default value for the "instances" usage param.  Without this, --sync-usage-file sets instances=0 causing the
+// costs for the node group to be $0.  This can be removed when --sync-usage-file creates the usage file with usgage keys
+// commented out by default.
+func (r *ElasticBeanstalkEnvironment) getUsageSchemaWithDefaultInstanceCount() []*schema.UsageItem {
+	var instanceCount = &r.instanceCount
+
+	if instanceCount == nil || *instanceCount == 0 {
+		return AutoscalingGroupUsageSchema
+	}
+
+	usageSchema := make([]*schema.UsageItem, 0, len(AutoscalingGroupUsageSchema))
+	for _, u := range AutoscalingGroupUsageSchema {
+		if u.Key == "instances" {
+			usageSchema = append(usageSchema, &schema.UsageItem{Key: "instances", DefaultValue: intVal(instanceCount), ValueType: schema.Int64})
+		} else {
+			usageSchema = append(usageSchema, u)
+		}
+	}
+	return usageSchema
+}
```

Comment:
I don't believe we need this as we're not using the `AutoscalingGroup` usage at the moment. I think this is something we could add down the line, but let's remove atm.